### ISSUE TITLE
How to add server_header_transformation in kind: EnvoyProxy

### ISSUE THREAD

**Issue by meektechie** [envoyproxy/gateway#4324]
*Description*:
>Describe the desired behavior, what scenario it enables and how it
would be used.
I want to remove server: nginx/1.27.1 header on on my response header. I am using envoy-gateway. I want server_header_transformation this to be applied globally.  
[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

Below i have attached my configuration. It did't works but need some experts guidelines.

`apiVersion: gateway.envoyproxy.io/v1alpha1
kind: EnvoyProxy
metadata:
  name: external-proxy-config
  namespace: gateway
spec:
  bootstrap:
    type: Merge
    value: |
      static_resources:
        listeners:
        - name: listener_0
          address:
            socket_address:
              address: 0.0.0.0
              port_value: 8080
          filter_chains:
          - filters:
            - name: envoy.filters.network.http_connection_manager
              typed_config:
                "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
                stat_prefix: ingress_http
                server_header_transformation: PASS_THROUGH
                server_name: none
                route_config:
                  name: local_route
                  virtual_hosts:
                  - name: local_service
                    domains: ["*"]
                    routes:
                    - match:
                        prefix: "/"
                      route:
                        cluster: some_service_cluster
        clusters:
        - name: some_service_cluster
          connect_timeout: 0.25s
          type: STRICT_DNS
          lb_policy: ROUND_ROBIN
          load_assignment:
            cluster_name: some_service_cluster
            endpoints:
            - lb_endpoints:
              - endpoint:
                  address:
                    socket_address:
                      address: 127.0.0.1
                      port_value: 19000
  provider:
    type: Kubernetes
    kubernetes:
      envoyDeployment:
        replicas: 1
        container:
          resources:
            requests:
              cpu: 150m
              memory: 640Mi
            limits:
              cpu: 500m
              memory: 1Gi
      envoyHpa:
        minReplicas: 1
        maxReplicas: 3
        metrics:
          - resource:
              name: cpu
              target:
                averageUtilization: 30
                type: Utilization
            type: Resource
      envoyService:
        externalTrafficPolicy: Local
        type: LoadBalancer
        annotations:
          service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: ip
          service.beta.kubernetes.io/aws-load-balancer-scheme: "internet-facing"`

**Comment by meektechie** [comment]
apiVersion: gateway.envoyproxy.io/v1alpha1
kind: EnvoyProxy
metadata:
  name: external-proxy-config
  namespace: gateway
spec:
  bootstrap:
    type: Merge
    value: |
      static_resources:
        listeners:
        - name: listener_0
          address:
            socket_address:
              address: 0.0.0.0
              port_value: 8080
          filter_chains:
          - filters:
            - name: envoy.filters.network.http_connection_manager
              typed_config:
                "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
                stat_prefix: ingress_http
                server_header_transformation: PASS_THROUGH
                server_name: none
                route_config:
                  name: local_route
                  virtual_hosts:
                  - name: local_service
                    domains: ["*"]
                    routes:
                    - match:
                        prefix: "/"
                      route:
                        cluster: some_service_cluster
        clusters:
        - name: some_service_cluster
          connect_timeout: 0.25s
          type: STRICT_DNS
          lb_policy: ROUND_ROBIN
          load_assignment:
            cluster_name: some_service_cluster
            endpoints:
            - lb_endpoints:
              - endpoint:
                  address:
                    socket_address:
                      address: 127.0.0.1
                      port_value: 19000
  provider:
    type: Kubernetes
    kubernetes:
      envoyDeployment:
        replicas: 1
        container:
          resources:
            requests:
              cpu: 150m
              memory: 640Mi
            limits:
              cpu: 500m
              memory: 1Gi
      envoyHpa:
        minReplicas: 1
        maxReplicas: 3
        metrics:
          - resource:
              name: cpu
              target:
                averageUtilization: 30
                type: Utilization
            type: Resource
      envoyService:
        externalTrafficPolicy: Local
        type: LoadBalancer
        annotations:
          service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: ip
          service.beta.kubernetes.io/aws-load-balancer-scheme: "internet-facing"

**Comment by arkodg** [comment]
@meektechie  its `PASS_THROUGH` by default https://github.com/envoyproxy/gateway/pull/2585/files

**Comment by meektechie** [comment]
@arkodg  Yes it is v1.1.1. I followed the doc https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto.html#envoy-v3-api-enum-extensions-filters-network-http-connection-manager-v3-httpconnectionmanager-serverheadertransformation. there it is mentioned default is OVERWRITE. I am new to it. I want to remove the "Server" header globally for all the HTTPRoutes. Pls suggest how can i proceed.

**Comment by arkodg** [comment]
@meektechie Envoy Proxy sets the default to OvERWRITE but Envoy Gateway sets the default to PASSTHROUGH

**Comment by meektechie** [comment]
Great, but still i could see the server header. Am i missing something?

**Comment by arkodg** [comment]
linking the use case here https://github.com/envoyproxy/gateway/issues/6671#issuecomment-3146495736  and marking it as help wanted
cc @sekar-saravanan

**Comment by HusseinKabbout** [comment]
We just hit this issue and would like to make `server_header_transformation` AND `server_name` configurable. However, I wouldn't suggest adding it to `EnvoyProxy` but to CTP where we also have a bunch of other HCM related configuration (e.g. timeouts).

@arkodg WDYT?

**Comment by guydc** [comment]
> We just hit this issue and would like to make server_header_transformation AND server_name configurable. However, I wouldn't suggest adding it to EnvoyProxy but to CTP where we also have a bunch of other HCM related configuration (e.g. timeouts).

+1, there's lots of header related options in CTP.
